AppBuilder/platform/views/viewComponent/ABViewGanttComponent.js

```diff
@@ -0,0 +1,567 @@
+         this.originalStartDate = currScale.start;
+         this.originalEndDate = currScale.end;
+
+         this.sort();
+      }, 10);
+   }
+
+   setScale(scale) {
+      const ganttElem = $$(this.ids.gantt);
+      if (!ganttElem) return;
+
+      const ganttData = ganttElem.getService("local");
+      if (!ganttData) return;
+
+      let newScales = [];
+
+      switch (scale) {
+         case "day":
+            newScales = [YEAR_SCALE, MONTH_SCALE, DAY_SCALE];
+            break;
+         case "week":
+            newScales = [YEAR_SCALE, MONTH_SCALE, WEEK_SCALE];
+            break;
+         case "month":
+            newScales = [YEAR_SCALE, MONTH_SCALE];
+            break;
+         case "year":
+            newScales = [YEAR_SCALE];
+            break;
+      }
+
+      const currScale = ganttElem.getService("local").getScales(),
+         start = webix.Date.add(this.originalStartDate, -1, scale, true),
+         end = webix.Date.add(this.originalEndDate, 1, scale, true);
+
+      ganttData.setScales(
+         start,
+         end,
+         currScale.precise,
+         currScale.cellWidth,
+         currScale.cellHeight,
+         newScales
+      );
+      ganttElem.$app.refresh();
+      ganttElem.getState().$batch({ top: 0, left: 0 });
+   }
+
+   /**
+    * @function show()
+    * Show this component.
+    */
+   onShow() {
+      const datacollection = this.view.datacollection;
+
+      this.datacollectionLoad(datacollection);
+      datacollection.loadData(0);
+
+      $$(this.ids.component)?.show();
+   }
+
+   sort() {
+      // TODO: sorting;
+      return;
+      // const gantt = $$(ids.gantt).getGantt();
+      // if (!gantt) return;
+
+      // // default sort
+      // const MAX_date = new Date(8640000000000000);
+      // gantt.sort(function(a, b) {
+      //    const aStartDate = a["start_date"],
+      //       aEndDate = a["end_date"],
+      //       aDuration = a["duration"] || 1,
+      //       bStartDate = b["start_date"],
+      //       bEndDate = b["end_date"],
+      //       bDuration = b["duration"] || 1;
+
+      //    // if no start date, then be a last item
+      //    if (
+      //       a[this.StartDateField.columnName] == null ||
+      //       b[this.StartDateField.columnName] == null
+      //    ) {
+      //       return (
+      //          (a[this.StartDateField.columnName] || MAX_date) -
+      //          (b[this.StartDateField.columnName] || MAX_date)
+      //       );
+      //    } else if (aStartDate != bStartDate) {
+      //       return aStartDate - bStartDate;
+      //    } else if (aEndDate != bEndDate) {
+      //       return aEndDate - bEndDate;
+      //    } else if (aDuration != bDuration) {
+      //       return bDuration - aDuration;
+      //    }
+      // }, false);
+   }
+
+   async taskAdd(taskData) {
+      const patch = this.convertValues(taskData);
+
+      try {
+         return await this.CurrentObject?.model().create(patch);
```

Comment:
We don't have any **Comments** on these methods so it is hard to visually know what to look at here.  So this would be a good time to **add comments**. 

With that said, is it expected that `taskAdd()` should return the value from the `.create()` operation?  

The other `.taskRemove()`  and `.taskUpdate()` don't do that.  

Is this the expected operations of those methods?